packages/grafana-ui/src/components/uPlot/context.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useContext } from 'react';
+import uPlot from 'uplot';
+import { DashboardCursorSync } from '@grafana/data';
+
+interface PlotContextType {
+  plot: uPlot | undefined;
+}
+
+/**
+ * @alpha
+ */
+export const PlotContext = React.createContext<PlotContextType>({} as PlotContextType);
+
+// Exposes uPlot instance and bounding box of the entire canvas and plot area
+export const usePlotContext = (): PlotContextType => {
+  return useContext<PlotContextType>(PlotContext);
+};
+
+export interface PlotSyncConfig {
+  key: string;
+  sync: DashboardCursorSync;
+}
+
+const PlotSyncContext = React.createContext<PlotSyncConfig | null>(null);
+
+interface PlotSyncContextProviderProps {
+  config: PlotSyncConfig;
+  children: React.ReactNode;
+}
+
+export function PlotSyncContextProvider({ config, children }: PlotSyncContextProviderProps) {
```

Comment:
Context wrapping entire dashboard (or anything that renders uPlot) for the sync config to be exposed

Reply:
very confused by the new context, thought we where to use PanelContext & the event bus, (which we will need to use these events in other non uPlot panels)